inst/tests/tests.Rraw

```diff
@@ -16163,23 +16163,106 @@ test(2107.3, names(DT), c('A','b','c'))
 setnames(DT, -(1:2), toupper)
 test(2107.4, names(DT), c('A','b','C'))
 
-# first and last should no longer load xts namespace, #3857
+# first and last should no longer load xts namespace, #3857, below commented test for interactive validation when xts present but not loaded or attached
+#stopifnot("xts"%in%installed.packages(), !"xts"%in%loadedNamespaces()); library(data.table); x=as.POSIXct("2019-01-01"); last(x); stopifnot(!"xts" %in% loadedNamespaces())
 x = as.POSIXct("2019-09-09")+0:1
 old = options(datatable.verbose=TRUE)
-test(2108.01, last(x), x[length(x)], notOutput="xts")
-test(2108.02, first(x), x[1L], notOutput="xts")
+test(2108.01, last(x), x[length(x)], output="!is.xts(x)")
+test(2108.02, first(x), x[1L], output="!is.xts(x)")
 if (test_xts) {
   xt = xts(1:2, x)
-  test(2108.03, last(xt, 2L), xt, output="last: using xts::last")
-  test(2108.04, first(xt, 2L), xt, output="first: using xts::first")
+  test(2108.03, last(xt, 2L), xt, output="using xts::last: is.xts(x)")
+  test(2108.04, first(xt, 2L), xt, output="using xts::first: is.xts(x)")
   xt = xts(matrix(1:4, 2L, 2L), x)
-  test(2108.05, last(xt, 2L), xt, output="last: using xts::last")
-  test(2108.06, first(xt, 2L), xt, output="first: using xts::first")
+  test(2108.05, last(xt, 2L), xt, output="using xts::last: is.xts(x)")
+  test(2108.06, first(xt, 2L), xt, output="using xts::first: is.xts(x)")
 }
 # first on empty df now match head(df, n=1L), #3858
 df = data.frame(a=integer(), b=integer())
-test(2108.11, first(df), df, notOutput="xts")
-test(2108.12, tail(df), df, notOutput="xts")
```

Comment:
this `tail(df)` was actually a mistake in old unit test, it was now amended to `last(df)` as it should be in the first place